src/main/java/appeng/container/implementations/ContainerFluidTerminal.java

```diff
@@ -0,0 +1,479 @@
+			{
+				// We can't push enough for the bucket
+				return;
+			}
+
+			IAEFluidStack inserted = Platform.poweredInsert( this.getPowerSource(), this.monitor, AEFluidStack.fromFluidStack( extract ), this.getActionSource() );
+			if( inserted != null && inserted.getStackSize() > 0 )
+			{
+				// Only try to extract the amount we DID insert
+				extract.amount -= Math.toIntExact( inserted.getStackSize() );
+			}
+
+			if( isBucket )
+			{
+				// Remove bucket and replace with EmptyBucket
+				player.inventory.setItemStack( new ItemStack( Items.BUCKET, 1 ) );
+			}
+			else
+			{
+				// Actually drain
+				fh.drain( extract, true );
+			}
+			this.updateHeld( player );
+		}
+	}
+
+	public void setTargetStack( final IAEFluidStack stack )
+	{
+		if( Platform.isClient() )
+		{
+			if( stack == null && this.clientRequestedTargetFluid == null )
+			{
+				return;
+			}
+			if( stack != null && this.clientRequestedTargetFluid != null && stack.getFluidStack().isFluidEqual( this.clientRequestedTargetFluid.getFluidStack() ) )
+			{
+				return;
+			}
+			NetworkHandler.instance().sendToServer( new PacketTargetFluidStack( (AEFluidStack) stack ) );
+		}
+
+		this.clientRequestedTargetFluid = stack == null ? null : stack.copy();
+	}
+
+	@Override
+	public void updateSetting( IConfigManager manager, Enum settingName, Enum newValue )
+	{
+		if( this.getGui() != null )
+		{
+			this.getGui().updateSetting( manager, settingName, newValue );
+		}
+	}
+
+	@Override
+	public void detectAndSendChanges()
+	{
+		if( Platform.isServer() )
+		{
+			if( this.monitor != this.terminal.getInventory( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) ) )
+			{
+				this.setValidContainer( false );
+			}
+
+			for( final Settings set : this.serverCM.getSettings() )
+			{
+				final Enum<?> sideLocal = this.serverCM.getSetting( set );
+				final Enum<?> sideRemote = this.clientCM.getSetting( set );
+
+				if( sideLocal != sideRemote )
+				{
+					this.clientCM.putSetting( set, sideLocal );
+					for( final IContainerListener crafter : this.listeners )
+					{
+						if( crafter instanceof EntityPlayerMP )
+						{
+							try
+							{
+								NetworkHandler.instance().sendTo( new PacketValueConfig( set.name(), sideLocal.name() ), (EntityPlayerMP) crafter );
+							}
+							catch( final IOException e )
+							{
+								AELog.debug( e );
+							}
+						}
+					}
+				}
+			}
+
+			if( !this.fluids.isEmpty() )
+			{
+				try
+				{
+					final IItemList<IAEFluidStack> monitorCache = this.monitor.getStorageList();
+
+					final PacketMEFluidInventoryUpdate piu = new PacketMEFluidInventoryUpdate();
+
+					for( final IAEFluidStack is : this.fluids )
+					{
+						final IAEFluidStack send = monitorCache.findPrecise( is );
+						if( send == null )
+						{
+							is.setStackSize( 0 );
+							piu.appendFluid( is );
+						}
+						else
+						{
+							piu.appendFluid( send );
+						}
+					}
+
+					if( !piu.isEmpty() )

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)